sourcegraph/cody
# Ghost text: Show on command driven selections

## Description

We primarily disabled the ghost text in this scenario because command driven selections were quite common, especially when searching text in VS Code - we wanted to avoid spamming the user.

However, we now only show the ghost text when:
1. There is a multiple line selection
2. There is a single line selection but all non-whitespace characters from this line are selected.

Those restrictions mean that we can re-enable the command driven selections without worrying about it spamming the user too much.

This fixes an issue where Vim keybindings would not trigger the ghost text.

## Test plan

With Vim:
1. Enable the `vscodevim.vim` extension
2. Select multiple lines
3. Check the ghost text shows

With search:
1. Search for a single symbol that isn't the only text on a line
2. Check ghost text doesn't show
